### PR TITLE
openjdk: explicitly specify build platform to avoid WSL autodetection shenanigans

### LINKED ISSUE #204152: `openjdk17` FTBFS on WSL2 specifically
```
       > checking for greadlink... [not found]
       > checking for readlink... /nix/store/a7gvj343m05j2s32xcnwr35v31ynlypr-coreutils-9.1/bin/readlink
       > checking for cygpath... [not found]
       > checking for wslpath... [not found]
       > checking for lsb_release... [not found]
       > checking for cmd.exe... [not found]
       > checking for cmp... /nix/store/j9p3g8472iijd50vhdprx0nmk2fqn5gv-diffutils-3.8/bin/cmp
       > checking for uniq... /nix/store/a7gvj343m05j2s32xcnwr35v31ynlypr-coreutils-9.1/bin/uniq
       > checking build system type... x86_64-pc-wsl
       > checking host system type... x86_64-pc-wsl
       > checking target system type... x86_64-pc-wsl
       > checking openjdk-build os-cpu... windows-x86_64
       > checking openjdk-target os-cpu... windows-x86_64
       > checking compilation type... native
       > checking Windows environment type... wsl1
       > configure: error: Incorrect wsl1 installation. Neither cygpath nor wslpath was found
       > configure exiting with result code 1
```

They're trying to be smart: https://github.com/openjdk/jdk17u-dev/blob/da6fca4d78465625775f7b384ba752185694f699/make/autoconf/build-aux/config.guess#L59

We should probably patch that check out, but maybe people with actual Java experience will have better ideas.

@edwtjo @asbachb

### PR DESCRIPTION
**Pull request by K900**
Fixes #204152

###### Description of changes

This is very stupid. I hate this.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by Mindavi — APPROVED** [review]

**Comment by Artturin** [comment]
should go to staging

**Comment by K900** [comment]
It's about 600 rebuilds most of which are just wrapping existing JAR files so should be fine for master.

File: pkgs/development/compilers/openjdk/13.nix [hunk 1]
```diff
@@ -74,6 +74,10 @@ let
       "--with-zlib=system"
       "--with-lcms=system"
       "--with-stdc++lib=dynamic"
+
+      # JDK's build-system attempts to specifically detect
+      # and special-case WSL1, and we don't want it to do that
+      "--build=${stdenv.buildPlatform.config}"
```
Comment (Artturin) [inline pkgs/development/compilers/openjdk/13.nix:80]: how about instead setting `configurePlatforms = [ "build" "host" ];` which will add `--build` and `--host`

Comment (K900) [reply pkgs/development/compilers/openjdk/13.nix]: That seems to work, thanks, applied

**Pull request merged**